subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaseCrossBuildResultsStore.java

```diff
@@ -36,57 +36,55 @@
 import java.util.Set;
 
 import static org.gradle.performance.results.ResultsStoreHelper.split;
+import static org.gradle.performance.results.ResultsStoreHelper.toList;
 
 public class BaseCrossBuildResultsStore<R extends CrossBuildPerformanceResults> implements ResultsStore, DataReporter<R>, Closeable {
 
     private final PerformanceDatabase db;
     private final String resultType;
 
     public BaseCrossBuildResultsStore(String resultType) {
-        this.db = new PerformanceDatabase("cross-build-results", new CrossBuildResultsSchemaInitializer());
+        this.db = new PerformanceDatabase("cross-build-results", new CrossBuildResultsSchemaInitializer(), new StaleDataCleanupInitializer());
         this.resultType = resultType;
     }
 
     @Override
     public void report(final R results) {
         try {
-            db.withConnection(new ConnectionAction<Void>() {
```

Comment:
This is also changing some inner class to lambda expression. I suggest reviewing with `Hide whitespace changes` checkbox enabled.

Reply:
Can you do the refactoring in a separate commit next time? So it is easier to spot the actual changes?

Reply:
I'll do it in a separate PR. Sorry for that.